### PR TITLE
Updated H5Oget_info* and H5Oinfo_t to be compatible with HDF5 1.12.0.

### PR DESCRIPTION
**Pull request by brtnfld**
Issue: build netcdf with hdf5-1.12.0 beta release fails in h5_test directory #1628

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Awesome! Thanks for doing this...

File: h5_test/tst_h_atts3.c [hunk 1]
```diff
@@ -148,8 +152,14 @@ main()
       for (i = 0; i < num_obj; i++)
       {
 	 /* Get the name, and make sure this is a type. */
+
+#if H5_VERSION_GE(1,12,0)
+	 if (H5Oget_info_by_idx3(grpid, ".", H5_INDEX_CRT_ORDER, H5_ITER_INC,
+				 i, &obj_info, H5O_INFO_BASIC, H5P_DEFAULT) < 0) ERR;
```
Comment (jhendersonHDF) [inline h5_test/tst_h_atts3.c]: Are there any places in the netCDF code that may use more of the information that gets returned here? If H5O_INFO_BASIC is enough, great. Just want to make sure that H5O_INFO_ALL wouldn't be more appropriate.

**Comment by jhendersonHDF** [comment]
Everything looks good to me; I just left a quick comment about usage of H5O_INFO_BASIC but otherwise this looks straightforward.

**Comment by WardF** [comment]
Thanks very much @brtnfld and all; reviewing this today as well as other PR's. Very much appreciated!

**Comment by edwardhartnett** [comment]
THis PR should be taken down. I have another I will put up shortly which includes these changes, merged with the current master and all the recent changes.

**Comment by brtnfld** [comment]
Closing PR, changes are going to be combined into another PR.

**Pull request closed without merging**